### DISCUSSION TITLE
Coldtype -> SVG -> AE -> Lottie

### DISCUSSION THREAD

**Discussion by andreamorales** [coldtype/coldtype#110] [Q&A]
Couldn't find more info in the docs on this. How do I export an animation as a series of SVGs that can then be input in AE and exported to an animation in Lottie? I think this would arguably be the lightest way to get type animation online.

**Comment by stenson** [comment]
I don't know a lot about the AE -> Lottie part of the process (since I've never used Lottie), but to generate svg's from a coldtype animation, you can specify the `fmt` kwarg on the `@animation` decorator, like this:

```python
from coldtype import *

@animation((540, 540), timeline=Timeline(20), fmt="svg")
def to_svg(f):
    return (StSt("A", Font.MutatorSans(), 250
        , wght=f.e())
        .align(f.a.r)
        .f(0))
```

Rendering this complete animation (by hitting the `a` key with the viewer app open) will output individual svg files for each frame, though I'm not sure if that something that can be understood by After Effects as something it could then export to Lottie.

From what I can tell, Lottie seems to work by describing animations with keyframes and high-level transformations (akin to how After Effects works or how Flash used to work). Unfortunately I'm not sure that meshes well with the Coldtype frame-by-frame-callback approach, which is more oriented towards frame-by-frame output formats (like gifs and video). So while Lottie might be able to render/play a Coldtype-generated animation, I think it might work just by showing and hiding each frame, rather than doing anything more high-level (and lighter-weight). That might still be lighter-weight than a video, since it'd be vector data, but it wouldn't be truly interpolating a type animation, for instance. (Of course I could be _way_ off about all of that, as I know very little about Lottie!)
